oximeter/oximeter/src/lib.rs

```diff
@@ -18,33 +20,111 @@
 //! sensor. A _target_ is the thing being measured -- the service or the DIMM, in these cases. Both
 //! targets and metrics may have key-value pairs associated with them, called _fields_.
 //!
-//! Motivating example
-//! ------------------
+//! Defining timeseries schema
+//! --------------------------
+//!
+//! Creating a timeseries starts by defining its schema. This includes the name
+//! for the target and metric, as well as other metadata such as descriptions,
+//! data types, and version numbers. Let's start by looking at an example:
+//!
+//! ```text
+//! [target]
+//! name = "foo"
+//! description = "Statistics about the foo server"
+//! authz_scope = "fleet"
+//! versions = [
+//!     { version = 1, fields = ["name", "id"],
+//! ]
+//!
+//! [[metrics]]
+//! name = "total_requests"
+//! description = "The cumulative number of requests served"
+//! datum_type = "cumulative_u64"
+//! units = "count"
+//! versions = [
+//!     { added_in = 1, fields = ["route", "method", "response_code"],
+//! ]
+//!
+//! [fields.name]
```

Comment:
Is it okay for this to come after the `[[metrics]]` section? Or, how do we define fields on the target vs fields on the metric(s)?

Reply:
In this TOML format, the fields are independent of the metrics and the target, so at the same "level" of the TOML hierarchy. The target and metric definitions refer to the fields by name. So I think they can go before or after, since we deserialize and then look up each field by name in the deserialized type.

Reply:
Ok. I've had weird experiences with table orderings and TOML before, but I can never remember the details.

Reply:
That's definitely true. Any array of tables always refers to the most recent entry. So if you have `[[foo]]` and then do `[[foo.bar]]`, that's a sub-table in the most recent entry in the `foo` array. There are probably other situations too, but thankfully they don't apply here!